packages/dd-trace/src/appsec/iast/analyzers/unvalidated-redirect-analyzer.js

```diff
@@ -35,28 +39,17 @@ class UnvalidatedRedirectAnalyzer extends InjectionAnalyzer {
     if (!value) return false
 
     const ranges = getRanges(iastContext, value)
-    return ranges && ranges.length > 0 && !this._areSafeRanges(ranges)
+    return ranges && ranges.length > 0 && this._hasUnsafeRange(ranges)
   }
 
-  // Do not report vulnerability if ranges sources are exclusively url,
-  // path params or referer header to avoid false positives.
-  _areSafeRanges (ranges) {
-    return ranges && ranges.every(
-      range => this._isPathParam(range) || this._isUrl(range) || this._isRefererHeader(range)
+  _hasUnsafeRange (ranges) {
+    return ranges && ranges.some(
```

Comment:
`ranges` is checked before the call. In any case, we could use optional chaining operator.